spec/lcm/integration/support/in_memory_ads.rb

```diff
@@ -0,0 +1,53 @@
+require 'sqlite3' unless RUBY_PLATFORM == 'java'
+
+module GoodData
+  class Datawarehouse
+    #  existence of Datawarehouse and InMemoryAds implementations is mutually exclusive
+    #  this is to make type checks pass, see InMemoryAds#class
+  end
+end
+
+module Support
+  class InMemoryAds
```

Comment:
Why don't we directly monkey-patch GoodData::Datawarehouse? I think it would eliminate the need for the extra "type-check code" and the global variable $in_memory_ads_instance.